src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2398,53 +2399,115 @@ def delete_ssl_cert(cmd, resource_group_name, certificate_thumbprint):
                  '\naz .. ssl import -n {1} -g {2} --key-vault-certificate-name {3} ' \
                  '--key-vault /subscriptions/[sub id]/resourceGroups/[rg]/providers/Microsoft.KeyVault/' \
-                 'vaults/{0}'.format(key_vault, name, resource_group_name, key_vault_certificate_name)
+                 'vaults/{0}'.format(key_vault, name, resource_group_name, cert)
         logger.warning(kv_msg)
         return
 
     kv_id_parts = parse_resource_id(kv_id)
     kv_name = kv_id_parts['name']
     kv_resource_group_name = kv_id_parts['resource_group']
     kv_subscription = kv_id_parts['subscription']
-    cert_name = '{}-{}-{}'.format(resource_group_name, kv_name, key_vault_certificate_name)
+    if not key_vault:
+        r = random.randrange(0, 99999)
+        cert_name = '{}-{}-{}'.format(cert, resource_group_name, r)
+    else:
+        cert_name = '{}-{}-{}'.format(resource_group_name, kv_name, cert)
     lnk = 'https://azure.github.io/AppService/2016/05/24/Deploying-Azure-Web-App-Certificate-through-Key-Vault.html'
     lnk_msg = 'Find more details here: {}'.format(lnk)
     if not _check_service_principal_permissions(cmd, kv_resource_group_name, kv_name, kv_subscription):
         logger.warning('Unable to verify Key Vault permissions.')
         logger.warning('You may need to grant Microsoft.Azure.WebSites service principal the Secret:Get permission')
         logger.warning(lnk_msg)
 
-    kv_cert_def = Certificate(location=location, key_vault_id=kv_id, password='',
-                              key_vault_secret_name=key_vault_certificate_name, server_farm_id=server_farm_id)
-
+    kv_cert_def = Certificate_Model(location=location, key_vault_id=kv_id, password='',
+                                    key_vault_secret_name=kv_secret_name, server_farm_id=server_farm_id)
     return client.certificates.create_or_update(name=cert_name, resource_group_name=resource_group_name,
                                                 certificate_envelope=kv_cert_def)
 
 
+def get_import_kv_details(cmd, client, certificate_name, key_vault, cert_subscription):
+    kv_id = None
+    kv_secret_name = certificate_name
+    # if key_vault is not populated, we are attempting to import an App Service Certificate
+    if not key_vault:
+        if cert_subscription.lower() != client.config.subscription_id.lower():
+            diff_subscription_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_APPSERVICE,
+                                                               subscription_id=cert_subscription)
+            certs = diff_subscription_client.app_service_certificate_orders.list()
+        else:
+            certs = client.app_service_certificate_orders.list()
+
+        cert = None
+        for c in certs:
+            if c.name == certificate_name:
+                cert = c.certificates[certificate_name]
+                break
+
```

Comment:
I am confused with whole code - Sorry! seems a lot a more complicated than I anticipated. Why are we assuming that KeyVault will not be populated the its an ASC? esp. since in the definition of the command key vault is required parameter. In general please add comments to the code to explain the logic or purpose so whoever takes over this in the future or is trying to investigate fixes will know what needs to be done.

Reply:
I am trying to mimic the portal functionality, since in portal we do not require the customer to provide KV name for app service certificates, as they don't need to. it seems easier on the customer end, but if we want to require it if we want the functionalities to match